pageserver/src/page_service.rs

```diff
@@ -3711,14 +3844,33 @@ impl proto::PageService for GrpcPageServiceHandler {
         req: tonic::Request<tonic::Streaming<proto::GetPageRequest>>,
     ) -> Result<tonic::Response<Self::GetPagesStream>, tonic::Status> {
         // Extract the timeline from the request and check that it exists.
+        //
+        // NB: during shard splits, the compute may still send requests to the parent shard. We'll
+        // reroute requests to the child shards below, but we also detect the common cases here
+        // where either the shard exists or no shards exist at all. If we have a child shard, we
+        // can't acquire a weak handle because we don't know which child shard to use yet.
+        //
+        // TODO: TimelineHandles.get() does internal retries, which will delay requests during shard
+        // splits. It shouldn't.
```

Comment:
@VladLazar What's your take on these internal retries? Should we keep them around? If so, I'll have to add a parameter to turn them off for the gRPC code paths, where they're actively harmful.